pkg/apis/autoscaling/v1alpha1/pa_validation.go

```diff
@@ -101,7 +124,14 @@ func (current *PodAutoscaler) CheckImmutableFields(ctx context.Context, og apis.
 		return &apis.FieldError{Message: "The provided original was not a PodAutoscaler"}
 	}
 
-	if diff, err := kmp.SafeDiff(original.Spec, current.Spec); err != nil {
+	// TODO(vagababov): remove after 0.6. This is temporary plug for backwards compatibility.
+	opt := cmp.FilterPath(
+		func(p cmp.Path) bool {
+			return p.String() == "ProtocolType" || p.String() == "Selector"
+		},
+		cmp.Ignore(),
```

Comment:
Consider ignoring these only when they are empty.

Reply:
I don't know how to write this filter... `FilterValues` has to be symmetric, while this would be different f(x, y) != f(y,x).

Reply:
I think you'd just replace `cmp.Ignore()` with:
```
cmp.Comparer(func(x, y interface{}) bool {
  // IsZero(x) || cmp.Equals(x,y)
})
```

Reply:
Not, symmetric :) will panic

Reply:
Since this is temporal, I think the way it is now should suffice, given that revisions are not editable and I'll add PA reconciliation in the followup PR